contracts/BentoBox.sol

```diff
@@ -207,9 +205,13 @@ contract BentoBox is MasterContractManager, BoringBatchable {
             "BentoBox: Skim too much"
         );
 
-        balanceOf[token][to] = balanceOf[token][to].add(share);
         total.base = total.base.add(share.to128());
+        if (total.elastic == 0) {
+            // Upon the first deposit permanently lock the MINIMUM_SHARE_BALANCE amount of tokens
+            share = share.sub(MINIMUM_SHARE_BALANCE);
```

Comment:
YieldBox has a different solution for this that doesn't burn any user tokens, but adds a virtual balance instead. Would love your review of this new method :)